components/vfs/src/root_file_system.cpp

```diff
@@ -198,13 +216,15 @@ StackErrorPtr RootFileSystem::enumerate_files(const std::string &file_path, cons
 		}
 	}
 
-	*files = files_with_extension;
+	*files = std::move(files_with_extension);
 
 	return nullptr;
 }
 
 StackErrorPtr RootFileSystem::enumerate_files_recursive(const std::string &file_path, const std::string &extension, std::vector<std::string> *files)
 {
+	assert(files);
```

Comment:
it looks like this will crash if it's run in release mode when asserts are presumably no-ops.  We might want to preemptively put in more graceful logic here.

Reply:
I will be revisiting the FS error handling to match the exception error handling that @asuessenbach suggested. I will fix this there